cpp/ql/src/Security/CWE/CWE-570/IncorrectAllocationErrorHandling.ql

```diff
@@ -16,6 +16,7 @@
 import cpp
 import semmle.code.cpp.valuenumbering.GlobalValueNumbering
 import semmle.code.cpp.controlflow.Guards
+import semmle.code.cpp.models.implementations.NoexceptFunction
```

Comment:
The import here is to facilitate the refactor with the `instanceof` check [on L175](https://github.com/github/codeql/pull/17298/files#diff-f2703f080209c49c8ca990c347ec96fb7ff33b008f8b1b179554df1f7a8ae099R175). It seems nice to me that the awareness of `noexcept` vs `throw()` is confined to `NoexceptFunction` compared to being duplicated several places.

Do you think this justifies the import or should I remove it (and undo the refactor) or structure it in some other way?

Reply:
Ah, I had read that use of `NoexceptFunction` as `NonThrowingFunction`.  It could probably be changed to the latter, but if you don't want to do that I'm happy with how it is now.